cow_instrument/InstrumentTree.h

```diff
@@ -17,30 +16,38 @@ class Detector;
  */
 class InstrumentTree {
 public:
-  InstrumentTree(std::unique_ptr<const Node> &&root, size_t nDetectors);
-
-  InstrumentTree(const InstrumentTree& other);
-
-  std::unique_ptr<NodeIterator> iterator() const;
+  InstrumentTree(std::vector<Node> &&nodes, size_t nDetectors);
+  InstrumentTree(const InstrumentTree& original) = default;
 
   const Node &root() const;
 
   const Detector &getDetector(size_t detectorIndex) const;
 
   unsigned int version() const;
 
-  std::unique_ptr<const InstrumentTree> modify(const Command &command) const;
+  InstrumentTree modify(size_t node, const Command &command) const;
+  InstrumentTree modify(const Node *node, const Command &command) const;
```

Comment:
the `size_t` node index is essential for enumerating over the `InstrumentTree`, but would be a bad interface for users who are after a specific Node. For `InstrumentTree::modify` I expect that the latter is normally true. `NodeIndex` is a nice idea, but other than being able to reduce the two signatures down to one, I'm not sure what it offers us?

Since we refactored this. We have removed the previous `Node::smartCopy` behaviour, which was used to select Components to modify. Now we simply execute the `Command` on that `Component`, which is better. However, I think we ought to have some way to `find` the node we want to modify. Are the node iterators on `InstrumentTree` enough? If so, maybe we should have modify take an iterator as an argument instead (or as well as) a `size_t` NodeIndex.